stout/flags.cc

```diff
@@ -18,63 +18,185 @@ namespace stout::flags {
+    } else {
+      if (field->options().HasExtension(stout::v1::subcommand)) {
+        std::cerr << "stout.v1.subcommand option should be annotated"
+                  << " on fields that are only inside 'oneof subcommand'"
+                  << std::endl;
+        exit(1);
+      }
 
-    for (const auto& name : flag.names()) {
-      AddOrExit(name, field, message);
-    }
+      const auto& flag = field->options().GetExtension(stout::v1::flag);
+
+      if (flag.names().empty()) {
+        std::cerr
+            << "Missing at least one flag name in 'names' for field '"
+            << field->full_name() << "'"
+            << std::endl;
+        std::exit(1);
+      }
+
+      if (flag.help().empty()) {
+        std::cerr
+            << "Missing flag 'help' for field '"
+            << field->full_name() << "'"
+            << std::endl;
+        std::exit(1);
+      }
+
+      for (const auto& name : flag.names()) {
+        auto [_, inserted] = fields_.emplace(name, field);
+
+        if (!inserted) {
+          std::cerr << "Encountered duplicate flag name "
+                    << "'" << name << "' for message '"
+                    << message->GetTypeName() << "'"
+                    << std::endl;
+          std::exit(1);
+        }
+      }
+
+      for (const auto& name : flag.deprecated_names()) {
+        auto [_, inserted] = fields_.emplace(name, field);
 
-    for (const auto& name : flag.deprecated_names()) {
-      AddOrExit(name, field, message);
+        if (!inserted) {
+          std::cerr << "Encountered duplicate (deprecated) flag name "
+                    << "'" << name << "' for message '"
+                    << message->GetTypeName() << "'"
+                    << std::endl;
+          std::exit(1);
+        }
+      }
     }
   }
 }
 
 ////////////////////////////////////////////////////////////////////////
 
-void Parser::AddOrExit(
-    const std::string& name,
-    const google::protobuf::FieldDescriptor* field,
-    google::protobuf::Message* message) {
-  auto [_, inserted] = fields_.emplace(name, field);
+Parser* Parser::TryLookupParserForSubcommand(const std::string& name) {
+  auto iterator = subcommand_fields_.find(name);
+  if (iterator != subcommand_fields_.end()) {
+    const google::protobuf::FieldDescriptor* field = iterator->second;
+    CHECK_NE(nested_parsers_.count(field), 0);
+    return nested_parsers_[field].get();
+  } else {
+    return nullptr;
+  }
+}
 
-  if (!inserted) {
-    std::cerr
-        << "Encountered duplicate flag name '" << name << "' "
-        << "for field '" << field->full_name() << "'"
-        << std::endl;
-    std::exit(1);
+////////////////////////////////////////////////////////////////////////
+
+google::protobuf::Message* Parser::GetMessageForField(
+    const google::protobuf::FieldDescriptor* field) {
+  if (field->containing_type() == standard_flags_->GetDescriptor()) {
+    return standard_flags_.get();
+  } else {
+    return CHECK_NOTNULL(message_);
```

Comment:
```suggestion
    return CHECK_NOTNULL(message_) << "Invalid 'message_' value";
```

Reply:
You can't add something like this to a `CHECK_NOTNULL`.